### PR TITLE
Add ability to specify channel overrides on the command line

### PR DESCRIPTION
**Pull request by mariusvniekerk**

### CODE REVIEW

File: conda_lock/conda_lock.py [hunk 1]
```diff
@@ -280,10 +280,13 @@ def make_lock_files(
             include_dev_dependencies=include_dev_dependencies,
         )
 
+        if not override_channels:
+            channels = lock_spec.channels
```
Comment (ericdill) [inline conda_lock/conda_lock.py:284]: you probably dont need this extra flag. I think you implicitly have this flag with `channels != ()`

Comment (ericdill) [reply conda_lock/conda_lock.py:284]: i.e., if any channels are provided to this function then you're overriding what's in the lock spec

File: conda_lock/conda_lock.py [hunk 2]
```diff
@@ -372,6 +375,23 @@ def parser():
...
+        """,
+        default=False,
+    )
+    parser.add_argument(
+        "-c",
+        "--channel",
+        nargs="?",
+        action="append",
+        help="Additional channels to include in the solve.  Requires the --override-channels flag",
+    )
```
Comment (ericdill) [inline conda_lock/conda_lock.py:393]: I think you need a default value here too or args.channels is gonna raise down on line 469

Comment (mariusvniekerk) [reply conda_lock/conda_lock.py:409]: changed the channels to allow None

File: conda_lock/conda_lock.py [hunk 3]
```diff
@@ -442,6 +481,8 @@ def main():
         platforms=args.platform,
         no_mamba=args.no_mamba,
         include_dev_dependencies=args.dev_dependencies,
+        channels=args.channels,
```
Comment (ericdill) [inline conda_lock/conda_lock.py:484]: delete this

**Review by ericdill — APPROVED** [review]

**Pull request merged**